javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -674,6 +674,44 @@ default int length() {
     @Override
     Map<K, V> replace(Tuple2<K, V> currentElement, Tuple2<K, V> newElement);
 
+    /**
+     * Replaces the entry for the specified key only if it is currently mapped to some value.
+
+     * @param key the key of the element to be substituted.
+     * @param value the new value to be associated with the key
+     * @return a new map containing key mapped to value if key was contained before. The old map otherwise.
+     */
+    default Map<K, V> replaceCurrentValue(K key, V value) {
+        Objects.requireNonNull(key, "key is null");
+        return this.get(key)
+                   .map(currentValue -> this.replace(Tuple.of(key, currentValue), Tuple.of(key, value)))
+                   .getOrElse(this);
```

Comment:
You chose the functional style, which is great. Here we can avoid the creation of several intermediate objects, which is more GC friendly:

```java
return containsKey(key) ? put(key, newValue) : this;
```

(Note: we omit `this.` most of the time)